x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportDeleteShutdownNodeAction.java

```diff
@@ -92,21 +92,15 @@ public void onFailure(Exception e) {
 
             @Override
             public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
+                listener.onResponse(AcknowledgedResponse.TRUE);
```

Comment:
I suggest completing the listener _after_ submitting the task to the reroute service, so that if there's ever a need to wait for the followup reroute to complete you can safely do so with `GET _cluster/health?wait_for_events=languid`.

Reply:
I suggest completing the listener after submitting the task to the reroute service, so that if there's ever a need to wait for the followup reroute to complete you can safely do so with `GET _cluster/health?wait_for_events=languid`.

(sorry for the noise, repeating my previous comment in a new review)